include/ruby/internal/fl_type.h

```diff
@@ -204,7 +209,7 @@ ruby_fl_type {
     RUBY_FL_SINGLETON = RUBY_FL_USER0,
 };
 
-enum { RUBY_FL_DUPPED = RUBY_T_MASK | RUBY_FL_EXIVAR | RUBY_FL_TAINT };
+enum { RUBY_FL_DUPPED = RUBY_T_MASK | RUBY_FL_EXIVAR | RUBY_FL_SHAREABLE };
```

Comment:
@ko1 is this intentional that `Object#dup` inherits shareability? Or should we delete it?

Reply:
(1) You can delete RUBY_FL_SHAREABLE  here.
(2) RUBY_FL_DUPPED and rb_dup_setup which uses RUBY_FL_DUPPED are not used in core and maybe nobody uses them, so we can remove these definitions.


```
$ gem-codesearch rb_dup_setup
/srv/gems/rhodes-7.1.17/platform/shared/ruby/include/ruby/ruby.h:#define DUPSETU
/srv/gems/rhodes-7.1.17/platform/shared/ruby/include/ruby/ruby.h:rb_dup_setup(VA
/srv/gems/ruby-compiler-0.1.1/vendor/ruby/include/ruby/ruby.h:#define DUPSETUP(d
/srv/gems/ruby-compiler-0.1.1/vendor/ruby/include/ruby/ruby.h:rb_dup_setup(VALUE
/srv/gems/zscan-2.0.9/ext/pack/ruby/internal/newobj.h:#define DUPSETUP   rb_dup_
/srv/gems/zscan-2.0.9/ext/pack/ruby/internal/newobj.h:rb_dup_setup(VALUE dup, VA
```

Reply:
OK, thanks for checking.

Reply:
https://github.com/ruby/ruby/pull/4100/commits/9e083a24e1405bb23efe6c98aef95db03ef06a7d resolves this.